src/test/java/api/loans/anonymization/AnonymizeLoansAfterXIntervalTests.java

```diff
@@ -361,6 +369,67 @@ public void testClosedLoansAnonymizedAfterIntervalPassed() {
         .getJson(), not(isAnonymized()));
   }
 
+  @Test
+  public void doNotAnonymizeLoansTwice() {
+    createAnonymizeAfterIntervalConfiguration(1, "minute");
+
+    final InventoryItemResource nod = itemsFixture.basedUponNod();
+    final InventoryItemResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+
+    final IndividualResource firstLoan = loansFixture.checkOutByBarcode(smallAngryPlanet, user);
+    final IndividualResource secondLoan = loansFixture.checkOutByBarcode(nod, usersFixture.rebecca());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet);
+    loansFixture.checkInByBarcode(nod);
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse firstAnonymization = anonymizeLoansInTenant();
+    assertThat(firstAnonymization.getAnonymizedLoans().size(), is(2));
+    assertThat(loansStorageClient.getById(firstLoan.getId()).getJson(), isAnonymized());
+    assertThat(loansStorageClient.getById(secondLoan.getId()).getJson(), isAnonymized());
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse secondAnonymization = anonymizeLoansInTenant();
+    assertThat(secondAnonymization.getAnonymizedLoans().size(), is(0));
+  }
+
+  @Test
+  public void ignoresAlreadyAnonymizedLoans() {
+    createAnonymizeAfterIntervalConfiguration(1, "minute");
+
+    final InventoryItemResource nod = itemsFixture.basedUponNod();
+    final InventoryItemResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final InventoryItemResource temeraire = itemsFixture.basedUponTemeraire();
+    final InventoryItemResource dunkirk = itemsFixture.basedUponDunkirk();
+
+    final IndividualResource firstLoan = loansFixture.checkOutByBarcode(smallAngryPlanet, user);
+    final IndividualResource secondLoan = loansFixture.checkOutByBarcode(nod, usersFixture.rebecca());
+    final IndividualResource thirdLoan = loansFixture.checkOutByBarcode(temeraire, usersFixture.james());
+    final IndividualResource fourthLoan = loansFixture.checkOutByBarcode(dunkirk, usersFixture.steve());
+
+    loansFixture.checkInByBarcode(smallAngryPlanet);
+    loansFixture.checkInByBarcode(nod);
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse firstAnonymization = anonymizeLoansInTenant();
+    assertThat(firstAnonymization.getAnonymizedLoans(), hasSize(2));
+    assertThat(loansStorageClient.getById(firstLoan.getId()).getJson(), isAnonymized());
+    assertThat(loansStorageClient.getById(secondLoan.getId()).getJson(), isAnonymized());
+
+    loansFixture.checkInByBarcode(temeraire);
+    loansFixture.checkInByBarcode(dunkirk);
+
+    setNextAnonymizationDateTime(ONE_MINUTE_AND_ONE);
+
+    LoanAnonymizationAPIResponse secondAnonymization = anonymizeLoansInTenant();
+    assertThat(secondAnonymization.getAnonymizedLoans(), hasSize(2));
```

Comment:
To confirm my understanding. Prior to this change, this second run would have reported that all 4 were annonymized (as that is the same list as processed loans)?

Reply:
Correct, the logic picked all the closed loans every time, even if they already anonymized.

The test wold return 4 loans if the old logic in place.